src/deploy/functions/services/authBlocking.ts

```diff
@@ -0,0 +1,127 @@
+import * as backend from "../backend";
+import * as identityPlatform from "../../../gcp/identityPlatform";
+import * as events from "../../../functions/events";
+import { FirebaseError } from "../../../error";
+
+const BEFORE_CREATE = events.v1.BEFORE_CREATE_EVENT || events.v2.BEFORE_CREATE_EVENT;
+
+/**
+ * Ensure that at most one blocking function of that type exists and merges identity platform options on our backend to deploy.
+ * @param endpoint the Auth Blocking endpoint
+ * @param wantBackend the backend we are deploying
+ */
+export function validateAuthBlockingTrigger(
+  endpoint: backend.Endpoint & backend.BlockingTriggered,
+  wantBackend: backend.Backend
+): void {
+  const blockingEndpoints = backend
+    .allEndpoints(wantBackend)
+    .filter((ep) => backend.isBlockingTriggered(ep)) as (backend.Endpoint &
+    backend.BlockingTriggered)[];
+  if (
+    blockingEndpoints.find(
+      (ep) =>
+        ep.blockingTrigger.eventType === endpoint.blockingTrigger.eventType && ep.id !== endpoint.id
+    )
+  ) {
+    throw new FirebaseError(
+      `Can only create at most one Auth Blocking Trigger for ${endpoint.blockingTrigger.eventType} events`
+    );
+  }
+}
+
+function configChanged(
+  newConfig: identityPlatform.BlockingFunctionsConfig,
+  config: identityPlatform.BlockingFunctionsConfig
+) {
+  if (
+    newConfig.triggers?.beforeCreate?.functionUri !== config.triggers?.beforeCreate?.functionUri ||
+    newConfig.triggers?.beforeSignIn?.functionUri !== config.triggers?.beforeSignIn?.functionUri
+  ) {
+    return true;
+  }
+  if (
+    !!newConfig.forwardInboundCredentials?.accessToken !==
+      !!config.forwardInboundCredentials?.accessToken ||
+    !!newConfig.forwardInboundCredentials?.idToken !==
+      !!config.forwardInboundCredentials?.idToken ||
+    !!newConfig.forwardInboundCredentials?.refreshToken !==
+      !!config.forwardInboundCredentials?.refreshToken
+  ) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Registers the auth blocking trigger to identity platform. On updates, we don't touch the options.
+ * @param endpoint the blocking endpoint
+ * @param update if this registration is an update
+ */
+export async function registerTrigger(
+  endpoint: backend.Endpoint & backend.BlockingTriggered,
+  update: boolean
+): Promise<void> {
+  const newBlockingConfig = await identityPlatform.getBlockingFunctionsConfig(endpoint.project);
+  const oldBlockingConfig = { ...newBlockingConfig };
```

Comment:
Since you're modifying `newBlockingConfig` and it's an interface, not a simple Record<string, string> I'd feel more comfortable if you used `utils.deepCopy`

Reply:
Changed to `cloneDeep()`